src/index.js

```diff
@@ -230,4 +225,51 @@ function strWithLen(len) {
   return new Array(len + 1).join( 'x' );
 }
 
-export { rules };
+function getTagName({ env, tagNameOption}) {
+  if (tagNameOption) {
+    return tagNameOption;
+  } else if (env === 'relay') {
+    return 'Relay.QL';
+  }
+
+  return 'gql';
+}
+
+function getConfig(context) {
+  const {
+    schemaJson, // Schema via JSON object
+    schemaJsonFilepath, // Or Schema via absolute filepath
+    env,
+    tagName: tagNameOption,
+  } = context.options[0];
+
+  return {
+    schemaJson,
+    schemaJsonFilepath,
+    env,
+    tagNameOption
+  }
+}
+
+const gqlProcessor = {
+  preprocess: function(text, filename) {
+    const { env, tagNameOption } = getConfig(globalContext);
```

Comment:
This `globalContext` variable might be unnecessary, and could be error-prone.

If instead you defined a special internal tag—for example, something like `ESLintPluginGraphQLFile`—and then adjusted the tag-matching logic on [line 100](https://github.com/apollostack/eslint-plugin-graphql/pull/26/files#diff-1fdf421c05c1140f6d71444ea2b27638R100) to accept that tag, you could avoid any global state and the use `getConfig` or `getTagName` here.